src/Core/BasePlugin.php

```diff
@@ -34,70 +34,70 @@ class BasePlugin implements PluginInterface
      *
      * @var bool
      */
-    protected $bootstrapEnabled = true;
+    protected bool $bootstrapEnabled = true;
 
     /**
      * Console middleware
      *
      * @var bool
      */
-    protected $consoleEnabled = true;
+    protected bool $consoleEnabled = true;
 
     /**
      * Enable middleware
      *
      * @var bool
      */
-    protected $middlewareEnabled = true;
+    protected bool $middlewareEnabled = true;
 
     /**
      * Register container services
      *
      * @var bool
      */
-    protected $servicesEnabled = true;
+    protected bool $servicesEnabled = true;
 
     /**
      * Load routes or not
      *
      * @var bool
      */
-    protected $routesEnabled = true;
+    protected bool $routesEnabled = true;
 
     /**
      * The path to this plugin.
      *
      * @var string
      */
-    protected $path;
+    protected string $path;
 
     /**
      * The class path for this plugin.
      *
      * @var string
      */
-    protected $classPath;
+    protected string $classPath;
 
     /**
      * The config path for this plugin.
      *
      * @var string
      */
-    protected $configPath;
+    protected string $configPath;
 
     /**
      * The templates path for this plugin.
      *
      * @var string
      */
-    protected $templatePath;
+    protected string $templatePath;
 
     /**
      * The name of this plugin
      *
      * @var string
      */
-    protected $name;
+    protected string $name;
```

Comment:
These should probably be nullable if we want to have more correct types.

Reply:
Changes them to nullable. Left the isset but it could also be !== null.